dandiapi/api/views/pagination.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import annotations
```

Comment:
It might be worth having a short comment blurb at the start of this file just explaining why this new `pagination` module exists. It's clear from the docstrings on each overridden class that they don't call `.count()`, but I think a top-level explanation on _why_ it's important that they don't call `.count()` would make this more readable.